### PR TITLE
Add decompilation test

### PR DESCRIPTION
**Pull request by gacholio**
Add a test for decompiling at exception catch when crossing an inlined
synchronized method.

Test both the success and failure (IllegalMonitorStateException) cases
when exitting the monitor for the inlined method.

Fixes: #9427

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
This cannot be tested/merged until all of the relevant fixes have been merged.

**Comment by DanHeidinga** [comment]
@gacholio has the JIT side of the fix gone in yet?  I'll wait for that before kicking testing

**Comment by gacholio** [comment]
No, this is not ready for test (probably not this week).

**Comment by gacholio** [comment]
jenkins test sanity plinux jdk8

**Comment by gacholio** [comment]
@DanHeidinga This is ready for merge (or more testing).

**Comment by gacholio** [comment]
From the test output:
```
17:35:21  Testing: decomp005
17:35:21  Test start time: 2020/05/11 18:33:56 Atlantic Standard Time
17:35:21  Running command: "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/openjdkbinary/j2sdk-image/bin/java"  -Xcompressedrefs -Xgcpolicy:gencon -Xjit:enableOSR,count=0  -Xdump -Xgcthreads1  -Xdump:system:events=abort  -Xjit:count=1,disableAsyncCompilation,limit={*.jit*},tryToInline={*.jitIn*} -agentlib:jvmtitest=test:decomp005 -cp "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" com.ibm.jvmti.tests.util.TestRunner
17:35:21  Time spent starting: 1 milliseconds
17:35:21  Time spent executing: 118 milliseconds
17:35:21  Test result: PASSED
17:35:21  Output from test:
17:35:21   [OUT] *** Testing [1/1]:	testInlinedSyncDecompile
17:35:21   [OUT] Exception caught in jitOuter: java.lang.InterruptedException
17:35:21   [OUT] *** Test took 10 milliseconds
17:35:21   [OUT] OK
17:35:21   [OUT] 
17:35:21  >> Success condition was found: [Return code: 0]
17:35:21  
17:35:21  Testing: decomp005-imse
17:35:21  Test start time: 2020/05/11 18:33:57 Atlantic Standard Time
17:35:21  Running command: "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/openjdkbinary/j2sdk-image/bin/java"  -Xcompressedrefs -Xgcpolicy:gencon -Xjit:enableOSR,count=0  -Xdump -Xgcthreads1  -Xdump:system:events=abort  -Xjit:count=1,disableAsyncCompilation,limit={*.jit*},tryToInline={*.jitIn*} -agentlib:jvmtitest=test:decomp005,args:imse -cp "/home/jenkins/workspace/Test_openjdk8_j9_sanity.functional_ppc64le_linux_Personal/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" com.ibm.jvmti.tests.util.TestRunner
17:35:21  Time spent starting: 0 milliseconds
17:35:21  Time spent executing: 126 milliseconds
17:35:21  Test result: PASSED
17:35:21  Output from test:
17:35:21   [OUT] *** Testing [1/1]:	testInlinedSyncDecompile
17:35:21   [OUT] Exception caught in jitOuter: java.lang.IllegalMonitorStateException
17:35:21   [OUT] *** Test took 11 milliseconds
17:35:21   [OUT] OK
17:35:21   [OUT] 
17:35:21  >> Success condition was found: [Return code: 0]
```

File: runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp005.c [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+	if (!ensureVersion(env, JVMTI_VERSION_1_1)) {
+		return JNI_ERR;
+	}   
+    
+	if (NULL != args) {
+		if (0 == strcasecmp(args, "imse")) {
+			imse = 1;
+		} else {
+			error(env, err, "Unknown option");
```
Comment (pshipton) [inline runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp005.c:54]: @gacholio this is failing to compile on Windows.
`warning C4700: uninitialized local variable 'err' used`

Comment (DanHeidinga) [reply runtime/tests/jvmtitests/src/com/ibm/jvmti/tests/decompResolveFrame/decomp005.c:54]: Created https://github.com/eclipse/openj9/pull/9544 to address the compile issue

**Pull request merged**